plugins/obs-ffmpeg/obs-ffmpeg.c

```diff
@@ -306,6 +306,8 @@ bool obs_module_load(void)
 	da_init(active_log_contexts);
 	da_init(cached_log_contexts);
 
+	if (pthread_mutex_init(&log_contexts_mutex, NULL) != 0)
+		return false;
```

Comment:
This seems unnecessary as the mutex is already initialized.

https://github.com/obsproject/obs-studio/blob/c938ea712bce0e9d8e0cf348fd8f77725122b9a5/plugins/obs-ffmpeg/obs-ffmpeg.c#L45

Reply:
Pretty sure `PTHREAD_MUTEX_INITIALIZER` is just used to initialize the value, so you can call for example `pthread_mutex_destroy()` on it safely,  Not to actually initialize the mutex itself.  I feel like it'd be a bit weird if you could initialize the actual mutex itself with a constant.

Reply:
Weird but possible. See _Static Initializers for Mutexes and Condition Variables_ at https://linux.die.net/man/3/pthread_mutex_init

Reply:
You either initialise the mutex with pthread_mutex_init(), or you use the static initialiser constant. Both is redundant. Normally you'd only use pthread_mutex_init() if you wanted to set some of the attributes to a non-default value. Since the call to pthread_mutex_init() is passing NULL for its attributes argument, the call can be removed.

Reply:
`obs_shutdown()` does not close the modules https://github.com/obsproject/obs-studio/blob/master/libobs/obs-module.c#L454  So the mutex will not be reset to PTHREAD_MUTEX_INITIALIZER again, and init in `obs_module_load()` is required.